angular-frontend/accelerator.yaml

```diff
@@ -0,0 +1,42 @@
+accelerator:
+  displayName: Angular Frontend
+  description: Angular Single Page Application with client side rendered web UI. Application provides Service to interact with backend API's.
+  tags:
+    - angular
+    - web
+    - tanzu
+
+  options:
+    - name: artifactId
+      inputType: text
+      defaultValue: "client-side-ui-starter"
+      label: Application artifact name
+      required: true
+    - name: backendService
+      inputType: text
+      defaultValue: "customer-profile-backend.namespace.svc.cluster.local"
+      label: Hostname of the backend services app. If backend services are deployed to the same cluster then a local route can be used (e.g. <backend-app>.<namespace>.svc.cluster.local).
+      required: true
+  imports:
+    - name: tap-workload
+
+engine:
+  let:
+    - name: backendServiceResourceName
+      expression: '#backendService.toLowerCase()'
+  chain:
+    - merge:
+        - include: [ "**" ]
+          exclude: [ "nginx.conf" ]
+        - include: [ "nginx.conf" ]
+          chain:
+            - type: ReplaceText
+              substitutions:
+                - text: "http://backend/api/"
+                  with: "'http://' + #backendServiceResourceName + '/api/'"
```

Comment:
Is it safe to assume that all backing services are going to have a URI template that includes "/api/"?  Is this too brittle of a pattern?  Should the accelerator require the full service URL?

Reply:
In the NGINX configuration forwarding is defined as:
```
location /api/ {
  proxy_pass http://backend/api/;
}
```

If '/api' is not included in the target location then NGINX will not add it. If services are deployed on another path then frontend developers will need to adapt the NGINX configuration.